### PR TITLE
Feat/store management

### PR DESCRIPTION
**Pull request by fredericvilcot**
This PR provides the way to build a `redux` store through the OKP4 Design System.

- Each declared store has its own builder which can be invoked anywhere out of the library.
- All builders are tested under unit tests
- All other test suites have been refactored to fully integrate those implementations

👉 The next step will be to export all necessary files to be bundled by `rollup` and then exported by the library.

### CODE REVIEW

File: src/domain/faucet/store/builder/store.builder.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    if (!this.isDependencies(dependencies)) {
+      throw new UnspecifiedError(
+        'Ooops... Dependencies must be provided to build a Faucet store...'
+      )
+    }
+    return new FaucetStoreBuilder({ ...this.faucetStoreParameters, dependencies })
+  }
+
+  public withEventBus(eventBus: DeepReadonly<EventBus>): FaucetStoreBuilder {
+    if (!(eventBus instanceof EventBus)) {
```
Comment (lolottetheclash) [inline src/domain/faucet/store/builder/store.builder.ts:38]: As we are using this instance check twice, maybe with could make a function of it? As `isDependencies `?

**Review by lolottetheclash — APPROVED** [review]
Nice work, well done !!!

File: src/domain/faucet/store/builder/store.builder.ts (hunk 1 above)
Comment (fredericvilcot) [reply src/domain/faucet/store/builder/store.builder.ts:38]: Here we use a simple instanceof based assertion so I'm not sure transforming it to a type-guard is relevant.
isDependencies type guard is much more relevant as it's harder to test 'faucetGateway' in value predicate directly in the if statement.

Comment (fredericvilcot) [reply src/domain/faucet/store/builder/store.builder.ts:38]: Yes you're right, but I think we could keep this assertion as existing, to mainly prevent from adding a method per assertion in each store builder, if you are OK with that

**Review by ccamel — APPROVED** [review]

**Review by JoeyDjr — APPROVED** [review]

**Pull request merged**